### DISCUSSION TITLE
Shell injection via PDF password support in ImageMagick 7

### DISCUSSION THREAD

**Discussion by cure53alexander** [ImageMagick/ImageMagick#2851] [Announcements]
Hi all. 
I discovered a vulnerability in ImageMagick regarding the handling of the "-authenticate" parameter, which allows to set a password for password protected PDF files. 
The user controlled password was not properly escaped/sanitized and it was therefore possible to inject additional shell commands. 
I reported this vulnerability to the ImageMagick team and after some back and forth via email, the issue was addressed correctly (given my own understanding of the code and the issue).
I created a blogpost describing the issue more in depth: https://insert-script.blogspot.com/2020/11/imagemagick-shell-injection-via-pdf.html

I want to thank the IM team to not only adressing this quickly but also allowing me give feedback regarding the ways to address this problem.

cheers,
Alex

**Comment by Beuc** [comment]
Hi, was this assigned a CVE ID?

**Reply by maxcrees** [reply]
It appears [CVE-2020-29599](https://nvd.nist.gov/vuln/detail/CVE-2020-29599) has been assigned to this issue.

**Reply by weidongkl** [reply]
CVE-2020-29599

**Reply by Crispy-fried-chicken** [reply]
Is there any fix commit?
